packages/contract-wrappers/src/contract_wrappers/erc721_token_wrapper.ts

```diff
@@ -0,0 +1,439 @@
+            normalizedOperatorAddress,
+            isApproved,
+            {
+                gas: txOpts.gasLimit,
+                gasPrice: txOpts.gasPrice,
+                from: normalizedOwnerAddress,
+            },
+        );
+        return txHash;
+    }
+    /**
+     * Enable or disable approval for a third party ("operator") to manage all of `ownerAddress`'s assets.
+     * Throws if `_tokenId` is not a valid NFT
+     * Emits the ApprovalForAll event.
+     * @param   tokenAddress    The hex encoded contract Ethereum address where the ERC721 token is deployed.
+     * @param   ownerAddress    The hex encoded user Ethereum address of the token owner.
+     * @param   operatorAddress The hex encoded user Ethereum address of the operator you'd like to set approval for.
+     * @param   isApproved      The boolean variable to set the approval to.
+     * @param   txOpts          Transaction parameters.
+     * @return  Transaction hash.
+     */
+    public async setProxyApprovalForAllAsync(
+        tokenAddress: string,
+        ownerAddress: string,
+        isApproved: boolean,
+        txOpts: TransactionOpts = {},
+    ): Promise<string> {
+        const proxyAddress = this._erc721ProxyWrapper.getContractAddress();
+        const txHash = await this.setApprovalForAllAsync(tokenAddress, ownerAddress, proxyAddress, isApproved, txOpts);
+        return txHash;
+    }
+    /**
+     * Set or reaffirm the approved address for an NFT
+     * The zero address indicates there is no approved address. Throws unless `msg.sender` is the current NFT owner,
+     * or an authorized operator of the current owner.
+     * Throws if `_tokenId` is not a valid NFT
+     * Emits the Approval event.
+     * @param   tokenAddress    The hex encoded contract Ethereum address where the ERC721 token is deployed.
+     * @param   approvedAddress The hex encoded user Ethereum address you'd like to set approval for.
+     * @param   tokenId         The identifier for an NFT
+     * @param   txOpts          Transaction parameters.
+     * @return  Transaction hash.
+     */
+    public async setApprovalAsync(
+        tokenAddress: string,
+        approvedAddress: string,
+        tokenId: BigNumber,
+        txOpts: TransactionOpts = {},
+    ): Promise<string> {
+        assert.isETHAddressHex('tokenAddress', tokenAddress);
+        assert.isETHAddressHex('approvedAddress', approvedAddress);
+        assert.isBigNumber('tokenId', tokenId);
+        const normalizedTokenAddress = tokenAddress.toLowerCase();
+        const normalizedApprovedAddress = approvedAddress.toLowerCase();
+
+        const tokenContract = await this._getTokenContractAsync(normalizedTokenAddress);
+        const tokenOwnerAddress = await tokenContract.ownerOf.callAsync(tokenId);
+        const txHash = await tokenContract.approve.sendTransactionAsync(normalizedApprovedAddress, tokenId, {
+            gas: txOpts.gasLimit,
+            gasPrice: txOpts.gasPrice,
```

Comment:
These params are not guarenteed to exist. Do `undefined` values get stripped?

Reply:
Yes. Later